docs/articles/documentation/test-api/intercepting-and-mocking-http-requests.md

```diff
@@ -0,0 +1,547 @@
+    }
+    ```
+
+* When a response is received, the hook starts preparing to call the `onResponse` method that handles the response.
+
+    At this moment, the hook processes settings that define whether to pass the response headers and body to the `onResponse` method. These settings are specified in the second constructor parameter. This parameter takes an object with two properties - `includeHeaders` and `includeBody`.
+
+    ```js
+    export default class RequestHook {
+        constructor (requestFilterRules, responseEventConfigureOpts) {
+            console.log(responseEventConfigureOpts.includeHeaders); // false
+            console.log(responseEventConfigureOpts.includeBody);    // false
+        }
+    ```
+
+* After all the necessary preparations are done, the `onResponse` method is called. This method is abstract in the base class. Override it in the descendant to handle the request sending.
+
+    ```js
+    onResponse (/*ResponseEvent event*/) {
+        throw new Error('Not implemented');
+    }
+    ```
+
+### Writing a Hook
+
+To sum up, this is what you need to write a custom hook.
+
+* inherit from the `RequestHook` class,
+* override the `onRequest` method to handle sending the request,
+* override the `onResponse` method to handle receiving the response.
+
+```js
+import { RequestHook } from 'testcafe';
+
+class MyRequestHook extends RequestHook {
+    constructor (requestFilterRules, responseEventConfigureOpts) {
+        super(requestFilterRules, responseEventConfigureOpts);
+        // ...
+    }
+    onRequest (event) {
+        // ...
+    }
+    onResponse (event) {
+        // ...
+    }
+}
+```
+
+The `onRequest` and `onResponse` methods receive an object that contains the event parameters.
+
+The `onRequest` method's `event` object exposes the following fields.
+
+Property | Type | Description
+-------- | ---- | --------------
+`requestOptions` | Object | Contains the request parameters. You can use it to change the request parameters before the request is sent.
+`isAjax`         | Boolean | Specifies if the request is performed using AJAX.
+
+The `requestOptions` object has the following properties.
+
+Property | Type | Description
+-------- | ---- | ------------
+`headers`     | Object  | The request headers in the property-value form.
+`body`        | [Buffer](https://nodejs.org/api/buffer.html) | The request body.
+`url`    | String | A URL to which the request is sent.
+`protocol` | String | The request protocol.
+`hostname` | String | The destination host name.
+`host`     | String | The destination host.
+`port`     | Number | The destination port.
+`path`     | String | The destination path.
+`method`   | String | The request method.
+`credentials` | Object | Credentials that were used to authenticate in the current session using NTLM or Basic authentication. For HTTP Basic authentication, these are `username` and `password`. NTLM authentication additionally specifies `workstation` and `domain`.
```

Comment:
It would be great to describe `credentials` option in detail or to add a link if it's possible.